src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -166,12 +166,15 @@ public function hydrate(object $component, array $data, string $componentName):
             }
 
             $value = $dehydratedValue;
+            $type = $property->getType() instanceof \ReflectionNamedType ? $property->getType() : null;
 
             if ($method = $liveProp->hydrateMethod()) {
                 // TODO: Error checking
                 $value = $component->$method($dehydratedValue);
-            } elseif ($property->getType() instanceof \ReflectionNamedType && !$property->getType()->isBuiltin()) {
-                $value = $this->normalizer->denormalize($value, $property->getType()->getName(), 'json', [self::LIVE_CONTEXT => true]);
+            } elseif (!$value && $type && $type->allowsNull() && is_a($type->getName(), \BackedEnum::class, true) && !\in_array($value, array_map(fn (\BackedEnum $e) => $e->value, $type->getName()::cases()))) {
```

Comment:
This feels a bit specific or heavy-handed, as we discussed. But I think we should do it... if nothing else to see if this fixes things or helps us discover more use-cases so we can get this just right.